pyspecdata/DCCT_function.py

```diff
@@ -249,85 +271,79 @@ def DCCT(
         A = this_nddata.C
         A.rename(a_shape.dimlabels[:-2][0], "smooshed")
 
+    # }}}
+    # {{{ vertical lines for coh paths
     def draw_span(
         ax1,
         ax2,
         label,
         this_label_num,
-        allow_for_text=allow_for_text_default,
-        allow_for_ticks=allow_for_ticks_default,
     ):
-        x1, y1 = ax1.transAxes.transform(r_[0, 0.95])
-        x2, y2 = ax2.transAxes.transform(r_[0, 0.05])
-        x1 -= allow_for_ticks
-        x_text = x1 - allow_for_text
-        x2 -= allow_for_ticks
-        # following line to create an offset for different dimension labels
-        label_spacing = this_label_num * label_spacing_multiplier
-        x1, y1 = fig.transFigure.inverted().transform(
-            r_[x1 - label_spacing, y1]
-        )
-        x_text, _ = fig.transFigure.inverted().transform(
-            r_[x_text - label_spacing, 0]
+        """Place the vertical lines and labels of the coherence transfer
+        pathways"""
+        label_spacing = (
+            this_label_num
+            + 1  # plus one so the first horizontal doesn't
+            #      overlap with the indirect axis
+        ) * vert_label_spacer  # depending on number of dims this will space
+        #                        the lines along x approp.
+        x1_disp = LHS_labels + bbox[0] - label_spacing  # x coord is the left
+        #                                                 side of the axis minus
+        #                                                 the spacing for
+        #                                                 text/ticks
+        # {{{ Take y coordinate of top and bottom of axes objects to get the 2
+        # points for drawing the lines. To be exact I pull this from the axes
+        # objects themselves.
+        _, y1_fig = (ax1.transAxes + fig.transFigure.inverted()).transform(
+            r_[0, 0.95]
         )
-        x2, y2 = fig.transFigure.inverted().transform(
-            r_[x2 - label_spacing, y2]
+        _, y2_fig = (ax2.transAxes + fig.transFigure.inverted()).transform(
+            r_[0, 0.05]
```

Comment:
taking y coordinates directly from axes objects since this is more straightforward than guessing the coordinates in figure coords